moveit_ros/planning/moveit_cpp/src/planning_component.cpp

```diff
@@ -195,7 +199,70 @@ PlanningComponent::PlanSolution PlanningComponent::plan(const PlanRequestParamet
   //    visual_tools_->publishRobotState(last_solution_trajectory_->getLastWayPoint(), rviz_visual_tools::TRANSLUCENT);
   //  }
   //}
-  return *last_plan_solution_;
+  return *last_plan_solution_;  // TODO: Make better
+}
+
+PlanningComponent::PlanSolution
+PlanningComponent::plan(const PlanningComponent::MultiPipelinePlanRequestParameters& parameters,
+                        std::function<bool(std::vector<std::thread> planning_threads,
+                                           PlanningComponent::MultiPipelinePlanRequestParameters& parameters)>
+                            stop_criterion_callback,
+                        std::function<PlanningComponent::PlanSolution(std::vector<PlanSolution> const& solutions)>
+                            solution_selection_callback)
+{
+  // Create solutions container
+  PlanningComponent::PlanSolutions planning_solutions{ parameters.multi_plan_request_parameters.size() };
+  std::vector<std::thread> planning_threads;
+  planning_threads.reserve(parameters.multi_plan_request_parameters.size());
+
+  // Launch planning threads
+  for (auto const& plan_request_parameter : parameters.multi_plan_request_parameters)
+  {
+    auto planning_thread = std::thread([&]() {
+      auto solution = plan(plan_request_parameter);
+      planning_solutions.pushBack(solution);
```

Comment:
some planners in OMPL (asymptotically optimal ones) can produce more than solution before exiting. These could also be retrieved and added to the `planning_solutions` in case the criteria the callback uses for selecting the best solution are different than the optimization objective used by planner.

Reply:
Do you know if multiple solutions are accessible through the interface that is provided by
```
PlanningComponent::PlanSolution PlanningComponent::plan(const PlanRequestParameters& parameters)
```

Reply:
Do you know if multiple solutions are accessible through the interface that is provided by
```
PlanningComponent::PlanSolution PlanningComponent::plan(const PlanRequestParameters& parameters)
```

Reply:
I don't know. My comment isn't that important. The way to get to these solutions is first check that `dynamic_cast<OMPLPlanningPipelinePtr>(planning_pipeline_ptr> != nullptr` (or something like that). Next, you have to get the ModelBasedPlanningContextPtr. Next, you loop over each solution in  `getOMPLSimpleSetup()->getProblemDefinition()->getSolutions()`. This can be done at some later point.